ITEvents/Screens/Event view/ShowMoreEventsView.swift

```diff
@@ -2,21 +2,39 @@ import Foundation
 import UIKit
 
 class ShowMoreEventsView: UIView {
-    @IBOutlet weak var showMoreEventsButton: UIButton!
+    @IBOutlet private weak var showMoreEventsButton: UIButton!
+    var showOrHideEventsAction: ((@escaping () -> Void) -> Void)?
+    private var descriptionsCount: Int = 0
+    private var collapsed: Bool = true
     
-    class func initiateAndSetup(with descriptionsCount: Int) -> ShowMoreEventsView {
+    class func initiateAndSetup(with descriptionsCount: Int, showOrHideEventsAction: ((@escaping () -> Void) -> Void)? = nil) -> ShowMoreEventsView {
         let showMoreEventsView: ShowMoreEventsView = SharedUtils.createPanelView(nibName: "ShowMoreEventsView")
-        let buttonTitle = ShowMoreEventsView.getDescriptionFor(hidden: true, count: descriptionsCount)
-        showMoreEventsView.showMoreEventsButton.setTitle(buttonTitle, for: .normal)
+        showMoreEventsView.showOrHideEventsAction = showOrHideEventsAction
```

Comment:
Instead of having all updates in static(class) function 
it'd better to have similar call `showMoreEventsView.setup(with descriptionsCount, ...)` here